### PR TITLE
Do not hardcode vendor bundles in webpack

### PR DESCRIPTION
**Pull request by xPaw**

### CODE REVIEW

File: webpack.config.js [hunk 1]
```diff
@@ -71,7 +60,13 @@ const config = {
 		json3: "JSON", // socket.io uses json3.js, but we do not target any browsers that need it
 	},
 	plugins: [
-		new webpack.optimize.CommonsChunkPlugin("js/bundle.vendor.js")
+		// socket.io uses debug, we don't need it
+		new webpack.NormalModuleReplacementPlugin(/debug/, path.resolve(__dirname, "scripts/noop.js")),
+		// automatically split all vendor dependancies into a separate bundle
+		new webpack.optimize.CommonsChunkPlugin({
+			name: "js/bundle.vendor.js",
+			minChunks: (module) => module.context && module.context.indexOf("node_modules") !== -1
```
Comment (astorije) [inline webpack.config.js]: How does this know what packages to build into the vendor bundle specifically?

Comment (xPaw) [inline webpack.config.js]: `node_modules` being in the path.

Comment (astorije) [inline webpack.config.js]: Right, that tells Webpack where are our packages, but is it going to load everything that's under `node_modules` (like lodash, that we don't use in the client)? Or is it the other way around, it discovers everything that's `require()`d, then puts them in the right file based on this line?

Comment (xPaw) [inline webpack.config.js]: It's based on requires. Basically it generates identical bundles to what master has now.

**Pull request merged**